wandb/data_types.py

```diff
@@ -411,6 +411,10 @@ def add_data(self, *data):
         # Add the new data
         self.data.append(data)
 
+        for item in data:
+            if isinstance(item, Image):
+                item._free_ram()
```

Comment:
It's a bit unfortunate we would need to do this for any data types that store data in memory.  Could we move this into the `Image`?  I was thinking we could reset the image in `to_data_array` assuming that's the last time we need attributes about the data itself.  This way we would only have one place to update other data types.

Reply:
Hmm - this is a good recommendation. In the next commit I removed this Table-specific check and put the `_free_ram` at the end of the constructor. The `to_data_array` is a user-facing function and rarely called. This way the default state of an Image is not to hold an in-memory copy of the data, however, after any access to `Image.image`, a copy is retained. We can move this pattern to other Media types easily as you pointed out. LMK what you think.